src/btree/bt_curnext.c

```diff
@@ -709,8 +722,9 @@ __wt_btcur_iterate_setup(WT_CURSOR_BTREE *cbt)
      */
     F_SET(cbt, WT_CBT_ITERATE_NEXT | WT_CBT_ITERATE_PREV);
 
-    /* Clear the count of deleted items on the page. */
-    cbt->page_deleted_count = 0;
+    /* Clear the count of deleted items and non-visible valid items on the page. */
```

Comment:
Update this comment to reflect the delete is globally visible?

Reply:
Fixed.